### PR TITLE
Fix `build --backend=xcode`

### PR DESCRIPTION
**Pull request by nacho4d**
Xcode now works for at least simple projects.
- Fixes exception when calling `meson build --backend=xcode`
- Fixes some typos and other minor indentation little issues when generating project.pbxproj file

### CODE REVIEW

File: mesonbuild/backend/xcodebackend.py [hunk 1]
```diff
@@ -24,9 +24,9 @@ class XCodeBackend(backends.Backend):
     def __init__(self, build):
         super().__init__(build)
         self.name = 'xcode'
-        self.project_uid = self.environment.coredata.guid.replace('-', '')[:24]
+        self.project_uid = self.environment.coredata.lang_guids['default'].replace('-', '')[:24]
         self.project_conflist = self.gen_id()
-        self.indent = '       '
+        self.indent = '	' # tab
```
Comment (nirbheek) [inline mesonbuild/backend/xcodebackend.py]: If this is a tab, maybe just use `'\t'`?

**Comment by nacho4d** [comment]
cc @jpakkane  @nirbheek  :)

File: mesonbuild/backend/xcodebackend.py (hunk 1 above)
Comment (nacho4d) [reply mesonbuild/backend/xcodebackend.py]: done 👍

**Comment by nacho4d** [comment]
@nirbheek Thank you for the info. I will do so :)

**Pull request merged**